mods/farming/api.lua

```diff
@@ -142,38 +151,93 @@ farming.place_seed = function(itemstack, placer, pointed_thing, plantname)
 		return
 	end
 
-	
 	-- return if any of the nodes is not registered
 	if not minetest.registered_nodes[under.name] then
 		return
 	end
 	if not minetest.registered_nodes[above.name] then
 		return
 	end
-	
+
 	-- check if pointing at the top of the node
 	if pt.above.y ~= pt.under.y+1 then
 		return
 	end
-	
+
 	-- check if you can replace the node above the pointed node
 	if not minetest.registered_nodes[above.name].buildable_to then
 		return
 	end
-	
+
 	-- check if pointing at soil
 	if minetest.get_item_group(under.name, "soil") < 2 then
 		return
 	end
-	
+
 	-- add the node and remove 1 item from the itemstack
 	minetest.add_node(pt.above, {name = plantname, param2 = 1})
+	tick(pt.above)
 	if not minetest.setting_getbool("creative_mode") then
 		itemstack:take_item()
 	end
 	return itemstack
 end
 
+farming.grow_plant = function(pos, elapsed)
+	local node = minetest.get_node(pos)
+	local name = node.name
+	local def = minetest.registered_nodes[name]
+
+	if not def.next_plant then
+		-- disable timer for fully grown plant
+		return
+	end
+
+	-- grow seed
+	if minetest.get_item_group(node.name, "seed") and def.fertility then
+		local soil_node = minetest.get_node_or_nil({x = pos.x, y = pos.y - 1, z = pos.z})
+		if not soil_node then
+			tick_again(pos)
+			return
+		end
+		-- omitted is a check for light, we assume seeds can germinate in the dark.
+		for _, v in pairs(def.fertility) do
+			if minetest.get_item_group(soil_node.name, v) ~= 0 then
```

Comment:
@sofar the problem with replying through email is that you don't see the edits people might have made.
I added `If another mod does it, it would be their fault.` shortly after posting it.